components/input-field/input-field.js

```diff
@@ -0,0 +1,46 @@
+class InputField extends React.Component {
+	constructor(props) {
+    super(props)
+
+    this.state = {
+      isRequired: false,
+    }
+  }
+  
+  shouldValidate = () => {
+    const { name } = this.props
+
+    if ((name === 'name') || (name === 'email')) {
```

Comment:
i'd make a required prop. This input field now can't be used in a form where the name is not required, also, the `isRequired` flag should not change during runtime, which makes it (IMHO) not a good fit for `state`